multiplier_analysis/nbconverted/1_get_multiplier_LV_coverage.py

```diff
@@ -49,265 +51,249 @@
-    specific_genes = ls_genes_out[ifile][1]
-    
-    # Only include those genes that are in multiplier otherwise will get NAs
-    generic_genes_processed, specific_genes_processed = process.process_generic_specific_gene_lists(
-        generic_genes, 
-        specific_genes, 
-        multiplier_model_z_processed
-    )
-    
-    # Randomly generate generic gene list that is the same size as specific gene list
-    generic_genes_processed_subset = random.sample(generic_genes_processed, len(specific_genes_processed))
-    print(len(generic_genes_processed_subset), len(specific_genes_processed))
-    
-    
-    generic_cov_i, specific_cov_i = process.get_LV_coverage(
-        generic_genes_processed_subset,
-        specific_genes_processed,
-        multiplier_model_z_processed
-    )
-    
-    generic_cov_matched.append(len(generic_cov_i)/num_significant_LVs)
-    specific_cov_matched.append(len(specific_cov_i)/num_significant_LVs)
-    
-gene_cov_matched = pd.DataFrame({'Proportion of significantly associated LVs covered': generic_cov_matched + specific_cov_matched,
-                         'gene type': ['generic']*len(ls_data_files) + ['specific']*len(ls_data_files) 
-                      })
+all_coverage = []
+for gene_label in dict_genes.keys():
+    merged_df = pd.DataFrame(
+        dict_nonzero_coverage[gene_label],
+        columns= ["nonzero LV coverage"]
+    ).merge(
+        pd.DataFrame(
+            dict_highweight_coverage[gene_label],
+            columns= ["highweight LV coverage"]
+        ),
+        left_index=True,
+        right_index=True
+    ) 
+    merged_df['gene type'] = gene_label
+    all_coverage.append(merged_df)
+
+all_coverage_df = pd.concat(all_coverage)
 
 
 # In[18]:
 
 
+all_coverage_df.head()
+
+
+# In[19]:
+
+
 # Plot coverage distribution given list of generic coverage, specific coverage
-print(generic_cov_matched)
-print(specific_cov_matched)
+nonzero_fig = sns.boxplot(
+    data=all_coverage_df, 
+    x='gene type',
+    y='nonzero LV coverage',
+    notch=True,
+    palette=['powderblue', 'grey']
+                         )
+plt.ylim(0, 700)
+nonzero_fig.set_xlabel("Gene Type",fontsize=14)
+nonzero_fig.set_ylabel(textwrap.fill("Number of LVs", width=30),fontsize=14)
+nonzero_fig.tick_params(labelsize=14)
+nonzero_fig.set_title("Number of LVs genes are present in", fontsize=16)
 
-import textwrap
-fig2 = sns.swarmplot(data=gene_cov_matched, 
-                  x='gene type', 
-                  y='Proportion of significantly associated LVs covered', 
-                  palette=['grey','powderblue'])
-fig2.set_xlabel("Gene Type",fontsize=14)
-fig2.set_ylabel(textwrap.fill("Proportion of pathway-associated LVs covered", width=30),fontsize=14)
-fig2.tick_params(labelsize=14)
-fig2.set_title("Coverage of pathway-associated LVs", fontsize=16)
 
+# In[20]:
+
+
+# Plot coverage distribution given list of generic coverage, specific coverage
+highweight_fig = sns.boxplot(data=all_coverage_df, 
```

Comment:
So the interpretation of these plots would be that generic vs. other genes have nonzero loading values for roughly the same number of LVs (i.e. the distributions are the same), but the other genes tend to contribute highly to the LVs more often than the generic genes?

Do you think the difference in this plot would be significant if you did, say, a t-test?

Reply:
Yes your interpretation is correct. I was also debating about adding a significance statistic here.